app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -109,14 +109,25 @@ def saml_response
     encode_response(
       current_user,
       name_id_format: name_id_format,
-      authn_context_classref: requested_authn_context,
+      authn_context_classref: ial_authn_context,
       reference_id: active_identity.session_uuid,
       encryption: current_service_provider.encryption_opts,
       signature: saml_response_signature_options,
       signed_response_message: current_service_provider.signed_response_message_requested,
     )
   end
 
+  def ial_authn_context
+    case requested_authn_context
+    when Array
```

Comment:
Can we just pluralize the method, always make it return an array? IMO methods that return things with different types, or singular methods that sometimes return arrays can be super confusing